### ISSUE TITLE
Products Ordered Report - Not grouped by product

### ISSUE THREAD

**Issue by nicholasscottfish** [magento/magento2#22087]
### Preconditions (*)
1. Magento 2.2.3

### Steps to reproduce (*)
1. In the Magento 2 admin go to Reports -> Products -> Ordered
2. Select a date range where you know you have sold products spanning across multiple orders
3. Select 'Show By' year
4. Hit refresh

### Expected result (*)
1. I would expect to see a list of products sold per year, grouped by product with a total quantity

### Actual result (*)
1. I see a list of products sold per year, with multiple entries for a single product on a per order basis

### Discussion
From what I can see in the source code, this still appears to be an issue in Magento 2.3.* and I can't see any related issues, if there are then please forgive me.

I have created a temporary patch file for this:

```php
diff --git a/Model/ResourceModel/Product/Sold/Collection.php b/Model/ResourceModel/Product/Sold/Collection.php
index 61dc77d1..8b91fdb2 100644
--- a/Model/ResourceModel/Product/Sold/Collection.php
+++ b/Model/ResourceModel/Product/Sold/Collection.php
@@ -66,7 +66,7 @@ class Collection extends \Magento\Reports\Model\ResourceModel\Order\Collection
         $this->getSelect()->reset()->from(
             ['order_items' => $this->getTable('sales_order_item')],
             [
-                'ordered_qty' => 'order_items.qty_ordered',
+                'ordered_qty' => 'SUM(order_items.qty_ordered)',
                 'order_items_name' => 'order_items.name',
                 'order_items_sku' => 'order_items.sku'
             ]
@@ -76,8 +76,10 @@ class Collection extends \Magento\Reports\Model\ResourceModel\Order\Collection
             []
         )->where(
             'order_items.parent_item_id IS NULL'
+        )->group(
+            'order_items.product_id'
         )->having(
-            'order_items.qty_ordered > ?',
+            'SUM(order_items.qty_ordered) > ?',
             0
         );
         return $this;

```

I will raise a PR if other people are experiencing this issue, any feedback would be welcome.

Regards,

Nick

**Comment by ghost** [comment]
Hi @nicholasscottfish thank you for your report, Feel free to propose the Pull request. thanks for collaboration :+1:

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99158`, `MAGETWO-99159` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by surabhisrivastava09** [comment]
I am working on this at #dmcdindia1

**Comment by nicholasscottfish** [comment]
@magento-engcom-team give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @nicholasscottfish. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @nicholasscottfish, here is your [Magento instance](https://i-22087-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-22087-2-2-develop.instances.magento-community.engineering/admin](https://i-22087-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
Hi @nicholasscottfish. Thank you for your report. 
The issue has been fixed in magento/magento2#22646 by @Surabhi-Cedcoss in 2.3-develop branch
Related commit(s):
 - [b8bc2885b4059fe6c0f932a26a6495d6f7399896](https://github.com/magento/magento2/commit/b8bc2885b4059fe6c0f932a26a6495d6f7399896)
 - [82d09dcee2e57733c454e34dd48cb322bcccc17b](https://github.com/magento/magento2/commit/82d09dcee2e57733c454e34dd48cb322bcccc17b)

The fix will be available with the upcoming 2.3.3 release.

**Comment by magento-engcom-team** [comment]
Hi @nicholasscottfish. Thank you for your report. 
The issue has been fixed in magento/magento2#23252 by @krishprakash in 2.2-develop branch
Related commit(s):
 - [446f0c0e6a7183328a8065e00aa843871c749e03](https://github.com/magento/magento2/commit/446f0c0e6a7183328a8065e00aa843871c749e03)
 - [c60aab5bad451da8b79e1c77f20cb24043a1b93e](https://github.com/magento/magento2/commit/c60aab5bad451da8b79e1c77f20cb24043a1b93e)

The fix will be available with the upcoming 2.2.10 release.
